crates/libs/bindgen/src/gen.rs

```diff
@@ -922,23 +922,23 @@ impl<'a> Gen<'a> {
                 _ => {
                     let name = self.param_name(param.def);
                     match param.array_info {
-                        ArrayInfo::Fixed(fixed) if fixed > 0 && self.reader.param_free_with(param.def).is_none() => {
-                            if self.reader.param_flags(param.def).output() {
-                                quote! { ::core::mem::transmute(::windows::core::as_mut_ptr_or_null(#name)), }
+                        ArrayInfo::Fixed(_) | ArrayInfo::RelativeLen(_) | ArrayInfo::RelativeByteLen(_) => {
+                            let flags = self.reader.param_flags(param.def);
+                            let map = if flags.optional() {
+                                quote! { #name.as_deref().map_or(::core::ptr::null(), |slice|slice.as_ptr()) }
                             } else {
-                                quote! { ::core::mem::transmute(::windows::core::as_ptr_or_null(#name)), }
-                            }
-                        }
-                        ArrayInfo::RelativeLen(_) => {
-                            if self.reader.param_flags(param.def).output() {
-                                quote! { ::core::mem::transmute(::windows::core::as_mut_ptr_or_null(#name)), }
-                            } else {
-                                quote! { ::core::mem::transmute(::windows::core::as_ptr_or_null(#name)), }
-                            }
+                                quote! { #name.as_ptr() }
+                            };
+                            quote! { ::core::mem::transmute(#map), }
```

Comment:
Since we know this is a pointer does `as _` work instead? Would be nice to avoid another use of transmute.

Reply:
Not just yet - it still requires a bit more untangling before that will work. There are a few cases where the underlying types aren't simply the pointer equivalent of the slice. But I'd love to get there.